### PR TITLE
check proposal templates when updating proposals

### PR DESCRIPTION
**Pull request by mattcasey**
POST /api/proposals - make sure the reviewers match on all evaluation steps, and that you use a template if space requires it
PUT /api/proposals/[id]/evaluation|rubric-critera - check if proposal came from a template

### CODE REVIEW

File: __integration-tests__/server/pages/api/proposals/index.spec.ts [hunk 1]
```diff
@@ -101,4 +153,88 @@ describe('POST /api/proposals - Create a proposal', () => {
     };
     await request(baseUrl).post('/api/proposals').set('Cookie', userCookie).send(input).expect(401);
   });
+
+  it('should succeed if the proposal matches the template and respond with 201', async () => {
```
Comment (mattcasey) [inline __integration-tests__/server/pages/api/proposals/index.spec.ts:157]: these are the new tests i added

File: pages/api/proposals/[id]/evaluation.ts [hunk 2]
```diff
@@ -52,8 +44,8 @@ async function updateEvaluationEndpoint(req: NextApiRequest, res: NextApiRespons
     throw error;
   }
 
-  // Only admins can update proposal templates
-  if (proposal.page?.type === 'proposal_template' && !isAdmin) {
+  // Only admins can update proposal templates or proposals made from a template
+  if ((proposal.page?.type === 'proposal_template' || proposal.page?.sourceTemplateId) && !isAdmin) {
```
Comment (mattcasey) [inline pages/api/proposals/[id]/evaluation.ts:48]: you shouldnt be able to update evaluation if it came from a template

File: pages/api/proposals/[id]/back-to-step.ts [hunk 3]
```diff
@@ -3,17 +3,13 @@ import nc from 'next-connect';
...
 import type { ReviewEvaluationRequest } from 'lib/proposal/submitEvaluationResult';
 import { withSessionRoute } from 'lib/session/withSession';
 
 const handler = nc<NextApiRequest, NextApiResponse>({ onError, onNoMatch });
 
-handler
-  .use(providePermissionClients({ key: 'id', location: 'query', resourceIdType: 'proposal' }))
-  .use(requireKeys(['evaluationId'], 'body'))
-  .put(updateEvaluationResultEndpoint);
+handler.use(requireKeys(['evaluationId'], 'body')).put(updateEvaluationResultEndpoint);
```
Comment (mattcasey) [inline pages/api/proposals/[id]/back-to-step.ts:12]: pretty sure we don't need this providePermissionClients middleware for proposal endpoints

File: pages/api/proposals/[id]/rubric-criteria.ts [hunk 4]
```diff
@@ -27,9 +55,17 @@ async function upsertProposalCriteriaController(
   if (!permissions.edit) {
     throw new ActionNotPermittedError(`You can't update this proposal.`);
   }
+  // Only admins can update proposal templates or proposals made from a template
+  if ((proposal.page?.type === 'proposal_template' || proposal.page?.sourceTemplateId) && !isAdmin) {
```
Comment (mattcasey) [inline pages/api/proposals/[id]/rubric-criteria.ts:59]: check template status when updating rubric criteria

File: pages/api/proposals/index.ts [hunk 5]
```diff
@@ -39,6 +39,55 @@ async function createProposalController(req: NextApiRequest, res: NextApiRespons
     if (!permissions.createProposals) {
       throw new ActionNotPermittedError('You cannot create new proposals');
     }
+    const space = await prisma.space.findUnique({
```
Comment (mattcasey) [inline pages/api/proposals/index.ts:42]: new logic when creating a proposal

File: __integration-tests__/server/pages/api/proposals/index.spec.ts [hunk 6]
```diff
@@ -101,4 +153,134 @@ describe('POST /api/proposals - Create a proposal', () => {
...
+          permissions: [],
+          index: 0,
+          rubricCriteria: []
+        }
+      ]
+    };
+    await request(baseUrl).post('/api/proposals').set('Cookie', userCookie).send(input).expect(401);
+  });
+
+  it('should pass if the proposal does not match the template and respond with 201', async () => {
```
Comment (motechFR) [inline __integration-tests__/server/pages/api/proposals/index.spec.ts:241]: Suggestion "it should succeed if the proposal does not match the template, but it is created by an admin, and respond 201"

I'd also do a quick sanity check on the actual proposal data to make sure the admin provided content was what was set, for both authors and reviewers

File: lib/proposal/createProposal.ts [hunk 7]
```diff
@@ -65,10 +60,7 @@ export async function createProposal({
   spaceId,
   pageProps,
   authors,
-  reviewers,
   evaluations = [],
-  evaluationType,
-  rubricCriteria,
   publishToLens,
```
Comment (motechFR) [inline lib/proposal/createProposal.ts:64]: Nice to see these fields go away

File: pages/api/proposals/index.ts [hunk 8]
```diff
@@ -18,15 +18,15 @@ handler.use(requireUser).post(createProposalController);
...
-        spaceId: proposalCreateProps.spaceId
-      }
-    });
+  const adminRole = await prisma.spaceRole.findFirst({
+    where: {
+      isAdmin: true,
+      userId: req.session.user.id,
+      spaceId: proposalCreateProps.spaceId
+    }
+  });
```
Comment (motechFR) [inline pages/api/proposals/index.ts:27]: There is a hasAccessToSpaceUtility method from core that gives the space role

File: pages/api/proposals/index.ts [hunk 9]
```diff
@@ -39,6 +39,58 @@ async function createProposalController(req: NextApiRequest, res: NextApiRespons
     if (!permissions.createProposals) {
       throw new ActionNotPermittedError('You cannot create new proposals');
     }
+    const space = await prisma.space.findUnique({
+      where: {
+        id: proposalCreateProps.spaceId
+      }
+    });
+    if (space?.requireProposalTemplate && !proposalCreateProps.pageProps?.sourceTemplateId) {
```
Comment (motechFR) [inline pages/api/proposals/index.ts:47]: Fair, good spot  👍

**Pull request merged**